pom.xml

```diff
@@ -271,6 +281,192 @@
+                                            <name>scala</name>
+                                        </property>
+                                        <property>
+                                            <name>groovy</name>
+                                        </property>
+                                        <property>
+                                            <name>javascript</name>
+                                        </property>
+                                        <property>
+                                            <name>json</name>
+                                        </property>
+                                        <property>
+                                            <name>kotlin</name>
+                                        </property>
+                                        <property>
+                                            <name>cpp</name>
+                                        </property>
+                                        <property>
+                                            <name>antlr4</name>
+                                        </property>
+                                        <property>
+                                            <name>sql</name>
+                                        </property>
+                                        <property>
+                                            <name>typescript</name>
+                                        </property>
+                                        <property>
+                                            <name>yaml</name>
+                                        </property>
+                                        <property>
+                                            <name>pom</name>
+                                        </property>
+                                        <property>
+                                            <name>markdown</name>
+                                        </property>
+                                        <property>
+                                            <name>gherkin</name>
+                                        </property>
+                                        <property>
+                                            <name>go</name>
+                                        </property>
+                                        <property>
+                                            <name>shell</name>
+                                        </property>
+                                    </nestedProperties>
+                                    <outputs>
+                                        <directories>
+                                            <directory>
+                                                <name>buildDir</name>
+                                            </directory>
+                                        </directories>
+                                        <cacheableBecause>check should only run when inputs change
```

Comment:
This could be said of all tasks that are cacheable. :) Let's add a more specific reason!

Reply:
I added something, however we could also update the sample with a better reason :)

Reply:
yes please do that as well!